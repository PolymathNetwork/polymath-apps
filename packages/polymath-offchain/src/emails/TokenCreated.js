// @flow

import React from 'react';

import { EmailWrapper, EtherscanURL } from './EmailWrapper';
import { POLYMATH_ISSUER_URL, POLYMATH_OFFCHAIN_URL } from '../constants';

type Props = {|
  ticker: string,
  txHash: string,
  networkId: string,
|};

// TODO @monitz87: check component and refactor if needed

export const TokenCreated = ({ txHash, ticker, networkId }: Props) => {
  return (
    <EmailWrapper>
      <h4>Congratulations!</h4>
      <h1>You Have Successfully Created the {ticker.toUpperCase()} Token</h1>
      <div className="icon-text tx-hash">
        <div className="icon">
          <img alt="Icon" src={`${POLYMATH_OFFCHAIN_URL}/img/checkmark.png`} />
        </div>
        <h2>You can view the transaction details here:</h2>
        <div className="tx">
          Transaction details on Etherscan:{' '}
          <EtherscanURL hash={txHash} type={'tx'} networkId={networkId} />
        </div>
      </div>
      <div className="text">
        <h3>
          Now that your security token is created, you can proceed to the next
          steps:
        </h3>
        <ol>
          <li>
            Mint tokens to existing shareholders and your reserve, or for
            investors that may have participated in a SAFT or pre-sale
            (optional).
          </li>
          <li>
            Whitelist investors after their suitability has been assessed.
            Investors can be added to or removed from the whitelist before,
            during and after the STO, allowing you to continuously add to your
            investor pool. Note that only investors whose ether addresses are
            included in the whitelist can participate into the STO or receive
            tokens.
          </li>
          <li>
            Select a security token offering template and configure the
            specifics of your security token offering.
          </li>
        </ol>
        <p align="center">
          <a href={`${POLYMATH_ISSUER_URL}/dashboard/${ticker}`}>
            <strong>Click here to proceed with your Token Minting</strong>
          </a>
        </p>
      </div>
    </EmailWrapper>
  );
};
