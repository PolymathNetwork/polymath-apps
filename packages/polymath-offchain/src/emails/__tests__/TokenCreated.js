import React from 'react';
import renderer from 'react-test-renderer';
import { TokenCreated } from '../TokenCreated';

describe('Component: TokenCreated', () => {
  test('renders with all props', () => {
    const component = renderer.create(
      <TokenCreated ticker={'SOMETICKER'} txHash={'0xffffffffffffffff'} />
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
