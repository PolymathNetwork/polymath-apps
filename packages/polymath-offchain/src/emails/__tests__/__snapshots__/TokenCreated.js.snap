// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TokenCreated renders with all props 1`] = `
<html
  lang="en"
>
  <head>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
  body {
    font: small/1.5 -apple-system, BlinkMacSystemFont, Arial, Helvetica, sans-serif;
    color: #152935;
    font-size: 16px;
  }
  
  a {
    color: #3D70B2 !important;
    text-decoration: none;
  }
  
  a:hover {
    text-decoration: underline;
  }
  
  ul,
  ol {
    margin-top: 5px;
  }
  
  ul li,
  ol li {
    font-size: 18px;
    margin-bottom: 7px;
  }
  
  ol li {
    padding-left: 5px;
  }
  
  h1 {
    font-size: 36px;
    line-height: 43px;
    font-weight: 600;
    color: #252E6A;
    margin-top: 0;
    margin-bottom: 11px;
  }
  
  h2 {
    font-size: 20px;
    line-height: 28px;
    color: #152935;
    font-weight: normal;
  }
  
  h3 {
    font-size: 20px;
    font-weight: 600;
    margin-bottom: 12px;
    color: #152935 !important;
  }
  
  h4 {
    color: #3D70B2;
    font-size: 14px;
    line-height: 18px;
    font-weight: 600;
    margin-top: 0;
    margin-bottom: 0;
  }
  
  .wrapper {
    width: 768px;
    margin: 0 auto;
  }
  
  .top-bar {
    height: 48px;
    border-bottom: 1px solid #DFE3E6;
    padding-left: 72px;
    margin-bottom: 24px;
  }
  
  .top-bar img {
    margin-top: 11px;
    width: 188px;
    height: 29px;
  }
  
  .content {
    padding: 3px 72px 25px;
  }
  
  .main-value {
    width: 624px;
    height: 160px;
    background-color: #EBF0F7;
    border-radius: 4px;
    margin-bottom: 32px;
    margin-top: 28px;
  }
  
  .main-value h2 {
    line-height: 18px;
    text-align: center;
    padding-top: 19px;
    font-weight: 600;
  }
  
  .main-value .value {
    width: 560px;
    height: 64px;
    line-height: 64px;
    text-align: center;
    font-size: 20px;
    background-color: white;
    margin: 0 auto;
    vertical-align: middle;
  }
  
  .icon-text {
    margin-bottom: 30px;
  }
  
  .icon-text .icon {
    width: 48px;
    height: 100%;
    float: left;
  }
  
  .icon-text .icon.question {
    margin-top: 5px;
  }
  
  .icon-text .icon img {
    width: 32px;
    height: 32px;
    margin-top: 4px;
  }
  
  .icon-text h2 {
    margin: 0;
    line-height: 25px;
  }
  
  .icon-text p {
    margin-top: 3px;
  }
  
  .icon-text.tx-hash {
    height: 57px;
    margin-top: 20px;
  }
  
  .text {
    margin-bottom: 26px;
    color: #5A6872;
  }
  
  .text p {
    font-size: 18px;
  }
  
  .value {
    margin-top: 13px;
  }
  
  .value p {
    margin-top: 8px;
  }
  
  .value strong {
    color: #152935 !important;
  }
  
  .footer {
    height: 48px;
    border-top: 1px solid #DFE3E6;
    padding-left: 72px;
    padding-right: 72px;
    font-size: 14px;
    color: #5A6872;
    padding-top: 14px;
  }
  
  .footer .left,
  .footer .right {
    float: left;
    width: 50%;
  }
  
  .footer .right {
    text-align: right;
  }
  
  .footer a {
    color: #5A6872 !important;
    margin-left: 40px;
  }
  
  .tx {
    float: left;
    width: 300px;
    height: 25px;
    border-radius: 4px;
    background-color: #EBF0F7;
    font-size: 12px;
    line-height: 25px;
    padding: 0 16px;
    margin-top: 6px;
    text-align: center;
  }
  
  .tx a {
    text-decoration: underline;
  }
  
  .sincere {
    font-weight: 600;
    color: #252E6A;
    line-height: 31px;
  }
",
        }
      }
    />
  </head>
  <body>
    <div
      className="wrapper"
    >
      <div
        className="top-bar"
      >
        <img
          alt="Icon"
          src="http://localhost:3001/img/logo.png"
        />
      </div>
      <div
        className="content"
      >
        <h4>
          Congratulations!
        </h4>
        <h1>
          You Have Successfully Created the 
          SOMETICKER
           Token
        </h1>
        <div
          className="icon-text tx-hash"
        >
          <div
            className="icon"
          >
            <img
              alt="Icon"
              src="http://localhost:3001/img/checkmark.png"
            />
          </div>
          <h2>
            You can view the transaction details here:
          </h2>
          <div
            className="tx"
          >
            Transaction details on Etherscan:
             
            <a
              href="https://local.etherscan.io/tx/0xffffffffffffffff"
              rel="noopener noreferrer"
              target="_blank"
            >
              0xffffffff...ffff
            </a>
          </div>
        </div>
        <div
          className="text"
        >
          <h3>
            Now that your security token is created, you can proceed to the next steps:
          </h3>
          <ol>
            <li>
              Mint tokens to existing shareholders and your reserve, or for investors that may have participated in a SAFT or pre-sale (optional).
            </li>
            <li>
              Whitelist investors after their suitability has been assessed. Investors can be added to or removed from the whitelist before, during and after the STO, allowing you to continuously add to your investor pool. Note that only investors whose ether addresses are included in the whitelist can participate into the STO or receive tokens.
            </li>
            <li>
              Select a security token offering template and configure the specifics of your security token offering.
            </li>
          </ol>
          <p
            align="center"
          >
            <a
              href="http://localhost:3000/dashboard/SOMETICKER"
            >
              <strong>
                Click here to proceed with your Token Minting
              </strong>
            </a>
          </p>
        </div>
        <div
          className="icon-text"
          style={
            Object {
              "height": "52px",
            }
          }
        >
          <div
            className="icon question"
          >
            <img
              alt="Icon"
              src="http://localhost:3001/img/question.png"
            />
          </div>
          <h2>
            If you have any questions, please reach out to
            <br />
            <a
              href="tokenstudio@polymath.network"
            >
              tokenstudio@polymath.network
            </a>
          </h2>
        </div>
        <h2
          className="sincere"
        >
          Best,
          <br />
          Polymath Support
        </h2>
      </div>
      <div
        className="footer"
      >
        <div
          className="left"
        >
          Â© 2018 Polymath
        </div>
        <div
          className="right"
        >
          <a
            href="https://polymath.network/termsofservice.html"
          >
            Terms of service
          </a>
          <a
            href="https://polymath.network/privacypolicy.html"
          >
            Privacy policy
          </a>
        </div>
      </div>
    </div>
  </body>
</html>
`;
